clip_image.py

```diff
@@ -0,0 +1,85 @@
+from typing import Optional, Iterable, Any, List
+
+import clip
+import numpy as np
+import torch
+from PIL import Image
+from jina import Executor, DocumentArray, requests
+
+
+def _batch_generator(data: List[Any], batch_size: int):
+    for i in range(0, len(data), batch_size):
+        yield data[i:min(i + batch_size, len(data))]
```

Comment:
There is no need to do this. A list can be indexed up to a position longer than it's length (in both start and end indices). As far as I understand ` data[i:min(i + batch_size, len(data))]` is the same as ` data[i: i + batch_size]`

Reply:
very nice catch! Thanks.